kore/src/Kore/Log.hs

```diff
@@ -321,6 +374,84 @@ makeKoreLogger exeName startTime timestampSwitch koreLogFormat logActionText =
 
     indent = Pretty.indent 4
 
+-- | The logger used by 'kore-rpc-booster' and 'kore-rpc-dev'
+makeKoreLogger ::
+    forall io.
+    MonadIO io =>
+    ExeName ->
+    KoreLogFormat ->
+    (Text -> Bool) ->
+    LogAction io Text ->
+    LogAction io Text ->
+    LogAction io SomeEntry
+makeKoreLogger exeName koreLogFormat entryFilter prettyLogAction jsonLogAction =
+    let actionForJsonLogs =
+            jsonLogAction
+                & Colog.cmap renderJson
+                & Colog.cfilter (entryFilter . entryTypeText)
+        actionForPrettyLogs =
+            prettyLogAction
+                & Colog.cmap render
+                & Colog.cfilter (not . entryFilter . entryTypeText)
+     in actionForJsonLogs <> actionForPrettyLogs
+  where
+    renderJson :: SomeEntry -> Text
+    renderJson (SomeEntry _context actualEntry) =
+        LazyText.toStrict . JSON.encodeToLazyText . addOriginField $ oneLineJson actualEntry
+      where
+        addOriginField :: JSON.Value -> JSON.Value
+        addOriginField = \case
+            (JSON.Object xs) -> JSON.Object $ JSON.insert (JSON.fromText "origin") (JSON.toJSON KoreRpc) xs
+            xs -> xs
+
+    render :: SomeEntry -> Text
+    render entry =
+        prettyActualEntry entry
+            & Pretty.layoutPretty Pretty.defaultLayoutOptions
+            & Pretty.renderText
+
+    exeName' = Pretty.pretty exeName <> Pretty.colon
+    prettyActualEntry entry@(SomeEntry entryContext actualEntry)
+        | OneLine <- koreLogFormat =
+            Pretty.hsep [header, oneLineDoc actualEntry]
+        | otherwise =
+            (Pretty.vsep . concat)
+                [ [header]
+                , indent <$> [longDoc actualEntry]
+                , context'
+                ]
+      where
+        header =
+            (Pretty.hsep . catMaybes)
+                [ Just exeName'
+                , Just severity'
+                , Just (Pretty.parens $ type' entry)
+                ]
+                <> Pretty.colon
+        severity' = prettySeverity (entrySeverity actualEntry)
+        type' e =
+            Pretty.pretty $
+                lookupTextFromTypeWithError $
+                    typeOfSomeEntry e
+        context' =
+            (entry : entryContext)
+                & reverse
+                & mapMaybe (\e -> (,type' e) <$> contextDoc e)
+                & prettyContext
+        prettyContext =
+            \case
+                [] -> []
+                xs -> ("Context" <> Pretty.colon) : (indent <$> mkContext xs)
+
+        mkContext =
+            \case
+                [] -> []
+                [(doc, typeName)] ->
+                    [Pretty.hsep [Pretty.parens typeName, doc]]
+                (doc, typeName) : xs -> (Pretty.hsep [Pretty.parens typeName, doc]) : (indent <$> (mkContext xs))
```

Comment:
i guess we will want to tweak this in a separate PR to resemble the booster contexts?

Reply:
Yes, certainly.